tapir/shifts/models.py

```diff
@@ -685,10 +685,90 @@ def is_balance_negative(self):
     def is_balance_positive(self):
         return self.get_account_balance() > 1
 
+    def get_current_shift_exemption(self, date=timezone.now().date()):
+        exemptions = ShiftExemption.objects.filter(shift_user_data=self).is_valid(date)
+        if exemptions.exists():
+            return exemptions.first()
+        return None
+
+    def is_currently_exempted_from_shifts(self, date=timezone.now().date()):
+        return (
+            ShiftExemption.objects.filter(shift_user_data=self).is_valid(date).exists()
+        )
+
 
 def create_shift_user_data(instance: TapirUser, **kwargs):
     if not hasattr(instance, "shift_user_data"):
         ShiftUserData.objects.create(user=instance)
 
 
 models.signals.post_save.connect(create_shift_user_data, sender=TapirUser)
+
+
+class ShiftExemption(models.Model):
+    shift_user_data = models.ForeignKey(
+        ShiftUserData, related_name="shift_exemptions", on_delete=models.CASCADE
+    )
+    start_date = models.DateField(_("Start date"), null=False, blank=False)
+    end_date = models.DateField(_("End date"), null=False, blank=False)
+    description = models.TextField(_("Description"), null=False, blank=False)
+
+    class ShiftExemptionQuerySet(models.QuerySet):
+        def is_valid(self, date=timezone.now().date()):
+            return self.filter(
+                start_date__lte=date,
+                end_date__gte=date,
+            )
+
+    objects = ShiftExemptionQuerySet.as_manager()
+
+    def is_valid(self, date=timezone.now().date()):
+        return self.start_date <= date <= self.end_date
+
+
+class ShiftCycleLog(models.Model):
```

Comment:
Hm, I still think that simply having a `CharfField` `user_key` or something on `ShiftAccountEntry` to guarantee idempotency would be the more low-tech and more general solution here.

If not then let's rename this to `ShiftCycleEntry` or something. Log is way overloaded as it is.

Reply:
The problem if we rely on the ShiftAccountEntry is that people that are exempted don't get one, so when we run the function again we'd need to check exemptions again. 
Or do we want to create ShiftAccountEntries with value 0 for exempted people? It would be a bit more explicit in the user's shift account log.

I'm not sure what you mean the user_key for? Do you mean a choice field with stuff like "manually_created", "shift_cycle_start", "shift_attended"...?